index.js

```diff
@@ -29,7 +29,7 @@ module.exports = {
         devDependencies: [
           'webpack.config.js',
           '**/__testUtils__/**/*.js',
-          '**/__tests__/*-test.js'
+          '**/__tests__/**/*-test.js'
```

Comment:
I think we should follow the lead of CRA here and support [the same filename conventions as they do](https://facebook.github.io/create-react-app/docs/running-tests#filename-conventions). That should also solve your issue. They also support the file extensions `jsx`, `ts` and `tsx` (see [source](https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/scripts/utils/createJestConfig.js#L41-L44)).

Therefore this would turn into:

```
devDependencies: [
  "webpack.config.js",
  "**/__testUtils__/**/*.{js,jsx,ts,tsx}",
  "**/__tests__/**/*.{js,jsx,ts,tsx}",
  "**/*.{spec,test}.{js,jsx,ts,tsx}"
]
```

That way this project would also get TypeScript support.

In the file `react.js` we need the same array of globs for configuring the overrides. Therefore I'd move this to a shared file `config.js` with a content like this:

```js
module.exports = {
  testFiles: [
    "**/__testUtils__/**/*.{js,jsx,ts,tsx}",
    "**/__tests__/**/*.{js,jsx,ts,tsx}",
    "**/*.{spec,test}.{js,jsx,ts,tsx}"
  ]
}
```

… and use that in `index.js`:

```js
devDependencies: config.testFiles.concat('webpack.config.js')
```

… and `react.js`:

```js
overrides: (base.overrides || []).concat({
  files: config.testFiles
  ...
}
```

I've tested this in a larger project which uses this config and it seems fine.